src/server.rs

```diff
@@ -1,209 +1,426 @@
+                paym_acc: pay_acc.clone(),
             },
-            wss: database.rpc().to_string(),
-            mul: properties.decimals,
-            recipient,
-            order,
-            result: match invoice.status {
-                InvoiceStatus::Unpaid(_) => "waiting",
-                InvoiceStatus::Paid(_) => "paid",
-            }
-            .into(),
-            version: env!("CARGO_PKG_VERSION").into(),
-        })
-    } else {
-        let tx = database.write()?;
-
-        tx.invoices()?.save(
-            &invoice_account,
-            &Invoice {
-                recipient: recipient_account,
-                order: order_encoded,
-                status: InvoiceStatus::Unpaid(price),
+        );
+
+        Ok((
+            OrderStatus {
+                order,
+                payment_status: PaymentStatus::Pending,
+                message: String::new(),
+                recipient: state.0.recipient.to_ss58check(),
+                server_info: ServerInfo {
+                    version: env!("CARGO_PKG_VERSION"),
+                    instance_id: String::new(),
+                    debug: state.0.debug,
+                    kalatori_remark: state.0.remark.clone(),
+                },
+                order_info: Some(OrderInfo {
+                    withdrawal_status: WithdrawalStatus::Waiting,
+                    amount,
+                    currency: CurrencyInfo {
+                        currency: "USDC".into(),
+                        chain_name: "assethub-polkadot".into(),
+                        kind: TokenKind::Assets,
+                        decimals: 6,
+                        rpc_url: state.rpc.clone(),
+                        asset_id: Some(1337),
+                    },
+                    callback,
+                    transactions: vec![],
+                    payment_account: pay_acc.to_ss58check(),
+                }),
             },
-        )?;
-
-        tx.commit()?;
-
-        Ok(Success {
-            pay_account: format!("0x{}", HexDisplay::from(&invoice_account.as_ref())),
-            price: price_f64,
-            wss: database.rpc().to_string(),
-            mul: properties.decimals,
-            recipient,
-            order,
-            version: env!("CARGO_PKG_VERSION").into(),
-            result: "waiting".into(),
-        })
+            OrderSuccess::Created,
+        ))
     }
 }
 
-fn convert(dec: u64, num: u128) -> Result<f64> {
-    #[allow(clippy::cast_precision_loss)]
-    let numfl = num as f64;
-    #[allow(clippy::cast_precision_loss)]
-    let mul = 10u128.pow(dec.try_into()?) as f64;
+async fn order(
+    state: extract::State<Arc<State>>,
+    matched_path: MatchedPath,
+    path_result: Result<RawPathParams, RawPathParamsRejection>,
+    query: Query<HashMap<String, String>>,
+) -> Response {
+    match process_order(state, &matched_path, path_result, &query).await {
+        Ok((order_status, order_success)) => match order_success {
```

Comment:
This doesn't work as specified in the API docs.
It **should** return the full details about the current status of the order (including amount, currency info etc), while the only thing I'm getting is

```json
{
    "message": "",
    "order": "123456",
    "payment_status": "unknown",
    "recipient": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
    "server_info": {
        "debug": true,
        "instance_id": "",
        "version": "0.2.0-rc1"
    }
}
```